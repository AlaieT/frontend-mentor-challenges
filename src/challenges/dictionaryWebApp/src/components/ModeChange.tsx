import React from "react";

import type { ModeChangeProps } from "../types";

import { ReactComponent as MoonIcon } from "../assets/images/icon-moon.svg";

import styles from "../styles/components/modeChange.module.scss";

const ModeChange = ({ callback = () => undefined, mode }: ModeChangeProps) => (
  <div id={styles.mode}>
    <label id={styles.switch} htmlFor={styles["mode-change"]}>
      <input
        id={styles["mode-change"]}
        type="checkbox"
        onChange={(e) => callback(e.target.checked ? "dark" : "light")}
      />
      <span className={styles.slider} />
    </label>
    <MoonIcon
      width={20}
      height={20}
      id={styles["moon-icon"]}
      className={mode === "dark" ? styles.dark : undefined}
    />
  </div>
);

export default ModeChange;
