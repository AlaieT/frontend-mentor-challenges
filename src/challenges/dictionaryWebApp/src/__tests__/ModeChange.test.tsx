import React from "react";
import {
  render,
  cleanup,
  fireEvent,
  screen,
  act
} from "@testing-library/react";

import ModeChange from "../components/ModeChange";

afterEach(cleanup);

describe("ModeChange", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(<ModeChange mode={"light"} />).asFragment()
      ).toMatchSnapshot();
    });
  });

  describe("correct functionality", () => {
    it("should be checked after click", async () => {
      const mockCallback = jest.fn();

      render(<ModeChange callback={mockCallback} mode={"light"} />);

      fireEvent.click(screen.getByRole("checkbox"));

      expect(screen.getByRole("checkbox")).toBeChecked();
      expect(mockCallback).toBeCalled();
    });
  });
});
