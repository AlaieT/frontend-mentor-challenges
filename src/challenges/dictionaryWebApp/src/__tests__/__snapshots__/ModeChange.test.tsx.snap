// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeChange correct render should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="mode"
  >
    <label
      for="mode-change"
      id="switch"
    >
      <input
        id="mode-change"
        type="checkbox"
      />
      <span
        class="slider"
      />
    </label>
    <div
      height="20"
      id="moon-icon"
      width="20"
    />
  </div>
</DocumentFragment>
`;
